openshiftcontrolplane/v1/types.go

```diff
@@ -46,8 +46,44 @@ type OpenShiftAPIServerConfig struct {
 
 	// TODO this needs to be removed.
 	APIServerArguments map[string][]string `json:"apiServerArguments"`
+
+	// apiServers holds information about enabled/disabled API servers
+	APIServers APIServers `json:"apiServers"`
+}
+
+type APIServers struct {
+	// perGroupOptions is a list of enabled/disabled API servers in addition to the defaults
+	PerGroupOptions []PerGroupOptions `json:"perGroupOptions"`
 }
 
+type PerGroupOptions struct {
+	// name is an API server name (see OpenShiftApiserverName
+	// typed constants for a complete list of available apiservers).
+	Name string `json:"name"`
+
+	// enabledVersions is a list of versions that must be enabled in addition to the defaults.
+	// Must not collide with the list of disabled versions
+	EnabledVersions []string `json:"enabledVersions"`
+
+	// disabledVersions is a list of versions that must be disabled in addition to the defaults.
+	// Must not collide with the list of enabled versions
+	DisabledVersions []string `json:"disabledVersions"`
+}
+
+type OpenShiftApiserverName string
+
+const (
+	OpenShiftAppsApiserver          OpenShiftApiserverName = "openshift.io/apps"
+	OpenShiftAuthorizationApiserver OpenShiftApiserverName = "openshift.io/authorization"
+	OpenShiftBuildApiserver         OpenShiftApiserverName = "openshift.io/build"
+	OpenShiftImageApiserver         OpenShiftApiserverName = "openshift.io/image"
+	OpenShiftProjectApiserver       OpenShiftApiserverName = "openshift.io/project"
+	OpenShiftQuotaApiserver         OpenShiftApiserverName = "openshift.io/quota"
+	OpenShiftRouteApiserver         OpenShiftApiserverName = "openshift.io/route"
+	OpenShiftSecurityApiserver      OpenShiftApiserverName = "openshift.io/security"
+	OpenShiftTemplateApiserver      OpenShiftApiserverName = "openshift.io/template"
+)
+
```

Comment:
How about matching the groups instead of making a new format?  So  apps.openshift.io for instance.